pinecone/core/grpc/index_grpc.py

```diff
@@ -183,10 +183,11 @@ def parse_query_response(response: dict):
     for match in response['results']:
         namespace = match.get('namespace', '')
         m = []
-        for item in match['matches']:
-            sc = ScoredVector(id=item['id'], score=item.get('score', 0.0), values=item.get('values', []),
-                              metadata=item.get('metadata', {}))
-            m.append(sc)
+        if 'matches' in match:
```

Comment:
Can you add unit tests for this function? Ideally for all our grpc response parsing functions here. Should help us catch future such issues sooner.